<?xml version="1.0" encoding="UTF-8"?>
<appendix id="logging_codes">
  <title>Logging codes</title>

  <section id="logging_codes_core">
    <title>Core: 000000 - 009999</title>

    <table frame="all">
      <title>Logging codes for core</title>
      <tgroup cols="3" align="left" colsep="1" rowsep="1">
        <colspec colname="c1" colwidth="*"/>
        <colspec colname="c2" colwidth="*"/>
        <colspec colname="c3" colwidth="6*"/>
        <thead>
          <row>
            <entry align="left">Code</entry>
            <entry align="left">Level</entry>
            <entry align="left">Description</entry>
          </row>
        </thead>
        <tbody>
          <row>
            <entry>100</entry>
            <entry><code>INFO</code></entry>
            <entry>Closing a connection for you. Please close them yourself</entry>
          </row>
          <row>
            <entry>102</entry>
            <entry><code>INFO</code></entry>
            <entry>Throwable trying to close a connection for you, please close it yourself</entry>
          </row>
          <row>
            <entry>103</entry>
            <entry><code>INFO</code></entry>
            <entry>Could not find a close method on alleged connection object. Please close your own connections</entry>
          </row>
          <row>
            <entry>151</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Some connections were not closed, see the log for the allocation stacktraces</entry>
          </row>
          <row>
            <entry>152</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Trying to return an unknown connection</entry>
          </row>
          <row>
            <entry>201</entry>
            <entry><code>ERROR</code></entry>
            <entry>SecurityContext setup failed</entry>
          </row>
          <row>
            <entry>202</entry>
            <entry><code>ERROR</code></entry>
            <entry>SecurityContext setup failed since CallbackSecurity was null</entry>
          </row>
          <row>
            <entry>251</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>SecurityContext setup failed</entry>
          </row>
          <row>
            <entry>252</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>SecurityContext setup failed since CallbackSecurity was null</entry>
          </row>
          <row>
            <entry>253</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Work is null</entry>
          </row>
          <row>
            <entry>254</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>StartTimeout is negative</entry>
          </row>
          <row>
            <entry>255</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Interrupted while requesting permit</entry>
          </row>
          <row>
            <entry>256</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Work execution context must be null because work instance implements WorkContextProviderStartTimeout is negative</entry>
          </row>
          <row>
            <entry>257</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Run method is synchronized</entry>
          </row>
          <row>
            <entry>258</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Release method is synchronized</entry>
          </row>
          <row>
            <entry>259</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unsupported WorkContext class</entry>
          </row>
          <row>
            <entry>260</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Duplicate TransactionWorkContext class</entry>
          </row>
          <row>
            <entry>261</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Duplicate SecurityWorkContext class</entry>
          </row>
          <row>
            <entry>262</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Duplicate HintWorkContext class</entry>
          </row>
          <row>
            <entry>263</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>WorkManager is shutting down</entry>
          </row>
          <row>
            <entry>264</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>SecurityContext setup failed since CallbackSecurity::Domain was empty</entry>
          </row>
          <row>
            <entry>265</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>ResourceAdapterAssociation failed</entry>
          </row>
          <row>
            <entry>266</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid number of parameters</entry>
          </row>
          <row>
            <entry>301</entry>
            <entry><code>INFO</code></entry>
            <entry>Registered a null handle for managed connection</entry>
          </row>
          <row>
            <entry>302</entry>
            <entry><code>INFO</code></entry>
            <entry>Unregistered handle that was not registered</entry>
          </row>
          <row>
            <entry>303</entry>
            <entry><code>INFO</code></entry>
            <entry>Unregistered a null handle for managed connection</entry>
          </row>
          <row>
            <entry>305</entry>
            <entry><code>WARN</code></entry>
            <entry>Connection error occured</entry>
          </row>
          <row>
            <entry>306</entry>
            <entry><code>WARN</code></entry>
            <entry>Unknown connection error occured</entry>
          </row>
          <row>
            <entry>307</entry>
            <entry><code>WARN</code></entry>
            <entry>Notified of error on a different managed connection</entry>
          </row>
          <row>
            <entry>311</entry>
            <entry><code>INFO</code></entry>
            <entry>Throwable from unregister connection</entry>
          </row>
          <row>
            <entry>312</entry>
            <entry><code>ERROR</code></entry>
            <entry>Error while closing connection handle</entry>
          </row>
          <row>
            <entry>313</entry>
            <entry><code>ERROR</code></entry>
            <entry>There is something wrong with the pooling</entry>
          </row>
          <row>
            <entry>314</entry>
            <entry><code>WARN</code></entry>
            <entry>Error during beforeCompletion</entry>
          </row>
          <row>
            <entry>315</entry>
            <entry><code>ERROR</code></entry>
            <entry>Pool has active handles</entry>
          </row>
          <row>
            <entry>316</entry>
            <entry><code>ERROR</code></entry>
            <entry>Handle allocation</entry>
          </row>
          <row>
            <entry>317</entry>
            <entry><code>ERROR</code></entry>
            <entry>Transaction boundary</entry>
          </row>
          <row>
            <entry>318</entry>
            <entry><code>ERROR</code></entry>
            <entry>Delisting resource in pool failed</entry>
          </row>
          <row>
            <entry>351</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Not correct type</entry>
          </row>
          <row>
            <entry>352</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Failure to delist resource</entry>
          </row>
          <row>
            <entry>353</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error in delist</entry>
          </row>
          <row>
            <entry>354</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unfinished local transaction - error getting local transaction</entry>
          </row>
          <row>
            <entry>355</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unfinished local transaction but managed connection does not provide a local transaction</entry>
          </row>
          <row>
            <entry>356</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Failed to enlist</entry>
          </row>
          <row>
            <entry>357</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error in dissociate</entry>
          </row>
          <row>
            <entry>401</entry>
            <entry><code>WARN</code></entry>
            <entry>Error during tidy up connection</entry>
          </row>
          <row>
            <entry>402</entry>
            <entry><code>WARN</code></entry>
            <entry>ResourceException in returning connection</entry>
          </row>
          <row>
            <entry>403</entry>
            <entry><code>WARN</code></entry>
            <entry>Reconnecting a connection handle that still has a managed connection</entry>
          </row>
          <row>
            <entry>404</entry>
            <entry><code>WARN</code></entry>
            <entry>Unchecked throwable in managedConnectionDisconnected()</entry>
          </row>
          <row>
            <entry>405</entry>
            <entry><code>WARN</code></entry>
            <entry>Multiple LocalTransaction connection listeners enlisted</entry>
          </row>
          <row>
            <entry>406</entry>
            <entry><code>ERROR</code></entry>
            <entry>Throwable in returning connection</entry>
          </row>
          <row>
            <entry>451</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The connection manager is shutdown</entry>
          </row>
          <row>
            <entry>452</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Method getManagedConnection retry wait was interrupted</entry>
          </row>
          <row>
            <entry>453</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to get managed connection</entry>
          </row>
          <row>
            <entry>454</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You are trying to use a connection factory that has been shut down: ManagedConnectionFactory is null</entry>
          </row>
          <row>
            <entry>455</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong ManagedConnectionFactory sent to allocateConnection</entry>
          </row>
          <row>
            <entry>456</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unchecked throwable in ManagedConnection.getConnection()</entry>
          </row>
          <row>
            <entry>457</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unchecked throwable in managedConnectionReconnected()</entry>
          </row>
          <row>
            <entry>458</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>This method is not supported</entry>
          </row>
          <row>
            <entry>459</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Transaction is not active</entry>
          </row>
          <row>
            <entry>460</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error checking for a transaction</entry>
          </row>
          <row>
            <entry>461</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Could not enlist in transaction on entering meta-aware object</entry>
          </row>
          <row>
            <entry>462</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Could not delist resource, probably a transaction rollback</entry>
          </row>
          <row>
            <entry>463</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to set XAResource transaction timeout</entry>
          </row>
          <row>
            <entry>464</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to find connection listener</entry>
          </row>
          <row>
            <entry>465</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Connection is null</entry>
          </row>
          <row>
            <entry>466</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Enlistment not enabled</entry>
          </row>
          <row>
            <entry>467</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Managed connection not lazy enlistable</entry>
          </row>
          <row>
            <entry>468</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Connection listener already enlisted</entry>
          </row>
          <row>
            <entry>469</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error during enlistment</entry>
          </row>
          <row>
            <entry>470</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You are trying to use a connection factory that has been shut down</entry>
          </row>
          <row>
            <entry>501</entry>
            <entry><code>WARN</code></entry>
            <entry>Thread is not the enlisting thread</entry>
          </row>
          <row>
            <entry>502</entry>
            <entry><code>WARN</code></entry>
            <entry>Transaction error in beforeCompletion</entry>
          </row>
          <row>
            <entry>503</entry>
            <entry><code>WARN</code></entry>
            <entry>Transaction error in afterCompletion</entry>
          </row>
          <row>
            <entry>504</entry>
            <entry><code>WARN</code></entry>
            <entry>Transaction not found</entry>
          </row>
          <row>
            <entry>601</entry>
            <entry><code>INFO</code></entry>
            <entry>ConnectionValidator has been interrupted</entry>
          </row>
          <row>
            <entry>602</entry>
            <entry><code>WARN</code></entry>
            <entry>ConnectionValidator ignored unexpected runtime exception</entry>
          </row>
          <row>
            <entry>603</entry>
            <entry><code>WARN</code></entry>
            <entry>ConnectionValidator ignored unexpected error</entry>
          </row>
          <row>
            <entry>604</entry>
            <entry><code>WARN</code></entry>
            <entry>Throwable while attempting to get a new connection</entry>
          </row>
          <row>
            <entry>605</entry>
            <entry><code>WARN</code></entry>
            <entry>Destroying connection that could not be successfully matched</entry>
          </row>
          <row>
            <entry>606</entry>
            <entry><code>WARN</code></entry>
            <entry>Throwable while trying to match managed connection, destroying connection</entry>
          </row>
          <row>
            <entry>607</entry>
            <entry><code>WARN</code></entry>
            <entry>ResourceException cleaning up managed connection</entry>
          </row>
          <row>
            <entry>608</entry>
            <entry><code>WARN</code></entry>
            <entry>Destroying returned connection, maximum pool size exceeded</entry>
          </row>
          <row>
            <entry>609</entry>
            <entry><code>WARN</code></entry>
            <entry>Attempt to return connection twice</entry>
          </row>
          <row>
            <entry>610</entry>
            <entry><code>WARN</code></entry>
            <entry>Unable to fill pool</entry>
          </row>
          <row>
            <entry>611</entry>
            <entry><code>WARN</code></entry>
            <entry>Warning: Background validation was specified with a non compliant ManagedConnectionFactory interface</entry>
          </row>
          <row>
            <entry>612</entry>
            <entry><code>WARN</code></entry>
            <entry>Destroying connection that could not be successfully matched</entry>
          </row>
          <row>
            <entry>613</entry>
            <entry><code>WARN</code></entry>
            <entry>Throwable while trying to match managed connection, destroying connection</entry>
          </row>
          <row>
            <entry>614</entry>
            <entry><code>ERROR</code></entry>
            <entry>Exception during createSubject()</entry>
          </row>
          <row>
            <entry>615</entry>
            <entry><code>WARN</code></entry>
            <entry>Destroying active connection in pool</entry>
          </row>
          <row>
            <entry>616</entry>
            <entry><code>ERROR</code></entry>
            <entry>Leak detected in pool</entry>
          </row>
          <row>
            <entry>617</entry>
            <entry><code>WARN</code></entry>
            <entry>Invalid incrementer capacity policy</entry>
          </row>
          <row>
            <entry>618</entry>
            <entry><code>WARN</code></entry>
            <entry>Invalid decrementer capacity policy</entry>
          </row>
          <row>
            <entry>619</entry>
            <entry><code>WARN</code></entry>
            <entry>Invalid capacity property</entry>
          </row>
          <row>
            <entry>620</entry>
            <entry><code>WARN</code></entry>
            <entry>Warning: ValidateOnMatch validation was specified with a non compliant ManagedConnectionFactory interface</entry>
          </row>
          <row>
            <entry>621</entry>
            <entry><code>WARN</code></entry>
            <entry>Destroying connection that could not be validated</entry>
          </row>
          <row>
            <entry>622</entry>
            <entry><code>WARN</code></entry>
            <entry>Unsupported pool implementation</entry>
          </row>
          <row>
            <entry>651</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to get managed connection pool</entry>
          </row>
          <row>
            <entry>652</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to obtain lock</entry>
          </row>
          <row>
            <entry>653</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The pool has been shutdown</entry>
          </row>
          <row>
            <entry>654</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Interrupted while requesting connection</entry>
          </row>
          <row>
            <entry>655</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>No managed connections available within configured blocking timeout</entry>
          </row>
          <row>
            <entry>656</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>This should never happen</entry>
          </row>
          <row>
            <entry>657</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Interrupted while requesting permit</entry>
          </row>
          <row>
            <entry>658</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unexpected throwable while trying to create a connection</entry>
          </row>
          <row>
            <entry>659</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to get connection listener</entry>
          </row>
          <row>
            <entry>701</entry>
            <entry><code>WARN</code></entry>
            <entry>Exception during unbind</entry>
          </row>
          <row>
            <entry>751</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Deployment failed</entry>
          </row>
          <row>
            <entry>851</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Resource adapter instance not active</entry>
          </row>
          <row>
            <entry>852</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Validation exception</entry>
          </row>
          <row>
            <entry>853</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The activation spec class is no longer available</entry>
          </row>
          <row>
            <entry>854</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The resource adapter is no longer available</entry>
          </row>
          <row>
            <entry>855</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Key isn't registered</entry>
          </row>
          <row>
            <entry>856</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to lookup resource adapter in MDR</entry>
          </row>
          <row>
            <entry>901</entry>
            <entry><code>WARN</code></entry>
            <entry>Error during connection close</entry>
          </row>
          <row>
            <entry>902</entry>
            <entry><code>ERROR</code></entry>
            <entry>Error during inflow crash recovery</entry>
          </row>
          <row>
            <entry>903</entry>
            <entry><code>ERROR</code></entry>
            <entry>Error creating Subject for crash recovery</entry>
          </row>
          <row>
            <entry>904</entry>
            <entry><code>WARN</code></entry>
            <entry>No security domain defined for crash recovery</entry>
          </row>
          <row>
            <entry>905</entry>
            <entry><code>WARN</code></entry>
            <entry>Subject for crash recovery was null</entry>
          </row>
          <row>
            <entry>906</entry>
            <entry><code>ERROR</code></entry>
            <entry>Error during crash recovery</entry>
          </row>
          <row>
            <entry>951</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error during connection close</entry>
          </row>
          <row>
            <entry>952</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error during recovery initialization</entry>
          </row>
          <row>
            <entry>953</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error during recovery shutdown</entry>
          </row>
          <row>
            <entry>1001</entry>
            <entry><code>WARN</code></entry>
            <entry>No users.properties were found</entry>
          </row>
          <row>
            <entry>1002</entry>
            <entry><code>ERROR</code></entry>
            <entry>Error while loading users.properties</entry>
          </row>
          <row>
            <entry>1003</entry>
            <entry><code>WARN</code></entry>
            <entry>No roles.properties were found</entry>
          </row>
          <row>
            <entry>1004</entry>
            <entry><code>ERROR</code></entry>
            <entry>Error while loading roles.properties</entry>
          </row>
          <row>
            <entry>1005</entry>
            <entry><code>WARN</code></entry>
            <entry>No callback.properties were found</entry>
          </row>
          <row>
            <entry>1006</entry>
            <entry><code>ERROR</code></entry>
            <entry>Error while loading callback.properties</entry>
          </row>
          <row>
            <entry>1011</entry>
            <entry><code>WARN</code></entry>
            <entry>Prepare called on a local tx. Use of local transactions on a JTA
            transaction with more than one branch may result in inconsistent data in some cases of failure</entry>
          </row>
          <row>
            <entry>1151</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Trying to start a new transaction when old is not complete</entry>
          </row>
          <row>
            <entry>1152</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Trying to start a new transaction with wrong flags</entry>
          </row>
          <row>
            <entry>1153</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error trying to start local transaction</entry>
          </row>
          <row>
            <entry>1154</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Throwable trying to start local transaction</entry>
          </row>
          <row>
            <entry>1155</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong xid in commit</entry>
          </row>
          <row>
            <entry>1156</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Could not commit local transaction</entry>
          </row>
          <row>
            <entry>1157</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Forget not supported in local transaction</entry>
          </row>
          <row>
            <entry>1158</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>No recovery for LocalTransaction only resource manager</entry>
          </row>
          <row>
            <entry>1159</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong xid in rollback</entry>
          </row>
          <row>
            <entry>1160</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Could not rollback local transaction</entry>
          </row>
        </tbody>
      </tgroup>
    </table>
  </section>

  <section id="logging_codes_common">
    <title>Common: 010000 - 019999</title>

    <table frame="all">
      <title>Logging codes for common</title>
      <tgroup cols="3" align="left" colsep="1" rowsep="1">
        <colspec colname="c1" colwidth="*"/>
        <colspec colname="c2" colwidth="*"/>
        <colspec colname="c3" colwidth="6*"/>
        <thead>
          <row>
            <entry align="left">Code</entry>
            <entry align="left">Level</entry>
            <entry align="left">Description</entry>
          </row>
        </thead>
        <tbody>
          <row>
            <entry>10001</entry>
            <entry><code>ERROR</code></entry>
            <entry>Parsing error of ra.xml file</entry>
          </row>
          <row>
            <entry>10002</entry>
            <entry><code>ERROR</code></entry>
            <entry>Parsing error of ironjacamar.xml file</entry>
          </row>
          <row>
            <entry>10003</entry>
            <entry><code>ERROR</code></entry>
            <entry>No @Connector was found and no definition in the ra.xml metadata either</entry>
          </row>
          <row>
            <entry>10004</entry>
            <entry><code>ERROR</code></entry>
            <entry>More than one @Connector was found but the correct one wasn't defined in the ra.xml metadata</entry>
          </row>
          <row>
            <entry>10051</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>AnnotationRepository reference is null</entry>
          </row>
          <row>
            <entry>10052</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>No @Connector defined</entry>
          </row>
          <row>
            <entry>10053</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>More than @Connector defined</entry>
          </row>
          <row>
            <entry>10054</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>More than one @ConnectionDefinitions defined</entry>
          </row>
          <row>
            <entry>10055</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unknown annotation</entry>
          </row>
          <row>
            <entry>10056</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Element isn't a valid boolean</entry>
          </row>
          <row>
            <entry>10057</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Attribute isn't a valid boolean</entry>
          </row>
          <row>
            <entry>10058</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Element isn't a valid number</entry>
          </row>
          <row>
            <entry>10059</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid flush strategy</entry>
          </row>
          <row>
            <entry>10060</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unexpected end tag</entry>
          </row>
          <row>
            <entry>10061</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unexpected element</entry>
          </row>
          <row>
            <entry>10062</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Reached end of xml document unexpectedly</entry>
          </row>
          <row>
            <entry>10063</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Mandatory class-name attribute missing</entry>
          </row>
          <row>
            <entry>10064</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unexpected attribute</entry>
          </row>
          <row>
            <entry>10065</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Missing mandatory jndi-name attribute</entry>
          </row>
          <row>
            <entry>10066</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You cannot define more than one pool or xa-pool in same connection-definition</entry>
          </row>
          <row>
            <entry>10067</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Element cannot be set without an xa-pool</entry>
          </row>
          <row>
            <entry>10068</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Missing required attribute</entry>
          </row>
          <row>
            <entry>10069</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Missing required element</entry>
          </row>
          <row>
            <entry>10070</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid negative value</entry>
          </row>
          <row>
            <entry>10071</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Tag is not valid</entry>
          </row>
          <row>
            <entry>10072</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Tag cannot be undefined</entry>
          </row>
          <row>
            <entry>10073</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid &lt;security&gt; configuration</entry>
          </row>
          <row>
            <entry>10074</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The resource adapter metadata must be defined</entry>
          </row>
          <row>
            <entry>10075</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The resource adapter metadata must contain either an outbound or inbound configuration</entry>
          </row>
          <row>
            <entry>10076</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Tag must be defined</entry>
          </row>
          <row>
            <entry>10077</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong annotation type</entry>
          </row>
          <row>
            <entry>10078</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid zero value</entry>
          </row>
        </tbody>
      </tgroup>
    </table>
  </section>

  <section id="logging_codes_deployers">
    <title>Deployers: 020000 - 029999</title>

    <table frame="all">
      <title>Logging codes for deployers</title>
      <tgroup cols="3" align="left" colsep="1" rowsep="1">
        <colspec colname="c1" colwidth="*"/>
        <colspec colname="c2" colwidth="*"/>
        <colspec colname="c3" colwidth="6*"/>
        <thead>
          <row>
            <entry align="left">Code</entry>
            <entry align="left">Level</entry>
            <entry align="left">Description</entry>
          </row>
        </thead>
        <tbody>
          <row>
            <entry>20001</entry>
            <entry><code>INFO</code></entry>
            <entry>Required license terms</entry>
          </row>
          <row>
            <entry>20002</entry>
            <entry><code>INFO</code></entry>
            <entry>Deployed</entry>
          </row>
          <row>
            <entry>20003</entry>
            <entry><code>WARN</code></entry>
            <entry>Failure during validation report generation</entry>
          </row>
          <row>
            <entry>20004</entry>
            <entry><code>WARN</code></entry>
            <entry>Only one connection definition found with a mismatch in class-name</entry>
          </row>
          <row>
            <entry>20005</entry>
            <entry><code>WARN</code></entry>
            <entry>Only one admin object found with a mismatch in class-name</entry>
          </row>
          <row>
            <entry>20006</entry>
            <entry><code>ERROR</code></entry>
            <entry>ConnectionFactory is null</entry>
          </row>
          <row>
            <entry>20007</entry>
            <entry><code>ERROR</code></entry>
            <entry>Exception during createSubject()</entry>
          </row>
          <row>
            <entry>20008</entry>
            <entry><code>WARN</code></entry>
            <entry>Invalid config property</entry>
          </row>
          <row>
            <entry>20009</entry>
            <entry><code>WARN</code></entry>
            <entry>Invalid connection definition</entry>
          </row>
          <row>
            <entry>20010</entry>
            <entry><code>ERROR</code></entry>
            <entry>Connection definition with missing class-name</entry>
          </row>
          <row>
            <entry>20011</entry>
            <entry><code>ERROR</code></entry>
            <entry>Admin object with missing class-name</entry>
          </row>
          <row>
            <entry>20012</entry>
            <entry><code>WARN</code></entry>
            <entry>Admin object not bound</entry>
          </row>
          <row>
            <entry>20013</entry>
            <entry><code>WARN</code></entry>
            <entry>Connection factory not bound</entry>
          </row>
          <row>
            <entry>20014</entry>
            <entry><code>INFO</code></entry>
            <entry>Admin object not spec compliant</entry>
          </row>
          <row>
            <entry>20015</entry>
            <entry><code>INFO</code></entry>
            <entry>Connection factory not spec compliant</entry>
          </row>
          <row>
            <entry>20016</entry>
            <entry><code>WARN</code></entry>
            <entry>Missing &lt;recovery&gt; element. XA recovery disabled</entry>
          </row>
          <row>
            <entry>20017</entry>
            <entry><code>WARN</code></entry>
            <entry>Invalid archive</entry>
          </row>
          <row>
            <entry>20018</entry>
            <entry><code>INFO</code></entry>
            <entry>Enabling &lt;validate-on-match&gt;</entry>
          </row>
          <row>
            <entry>20019</entry>
            <entry><code>INFO</code></entry>
            <entry>Changed TransactionSupport</entry>
          </row>
          <row>
            <entry>20051</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to start</entry>
          </row>
          <row>
            <entry>20052</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to associate</entry>
          </row>
          <row>
            <entry>20053</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>ManagedConnectionFactory must be defined in class-name</entry>
          </row>
          <row>
            <entry>20054</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>AdminObject must be defined in class-name</entry>
          </row>
          <row>
            <entry>20055</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Failed to bind admin object</entry>
          </row>
          <row>
            <entry>20056</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Deployment failed</entry>
          </row>
          <row>
            <entry>20057</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid ManagedConnectionFactory class</entry>
          </row>
          <row>
            <entry>20058</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid ActivationSpec class</entry>
          </row>
          <row>
            <entry>20059</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid ResourceAdapter class</entry>
          </row>
          <row>
            <entry>20060</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to inject</entry>
          </row>
          <row>
            <entry>20061</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid required work context</entry>
          </row>
          <row>
            <entry>20062</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid connection factory interface</entry>
          </row>
          <row>
            <entry>20063</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid connection factory implementation</entry>
          </row>
          <row>
            <entry>20064</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid connection interface</entry>
          </row>
          <row>
            <entry>20065</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid connection implementation</entry>
          </row>
          <row>
            <entry>20066</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Connection factory implementation doesn't implement interface</entry>
          </row>
          <row>
            <entry>20067</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Connection implementation doesn't implement interface</entry>
          </row>
        </tbody>
      </tgroup>
    </table>
  </section>

  <section id="logging_codes_adapters">
    <title>Adapters: 030000 - 039999</title>

    <table frame="all">
      <title>Logging codes for adapters</title>
      <tgroup cols="3" align="left" colsep="1" rowsep="1">
        <colspec colname="c1" colwidth="*"/>
        <colspec colname="c2" colwidth="*"/>
        <colspec colname="c3" colwidth="6*"/>
        <thead>
          <row>
            <entry align="left">Code</entry>
            <entry align="left">Level</entry>
            <entry align="left">Description</entry>
          </row>
        </thead>
        <tbody>
          <row>
            <entry>30000</entry>
            <entry><code>WARN</code></entry>
            <entry>Unable to load connection listener</entry>
          </row>
          <row>
            <entry>30001</entry>
            <entry><code>WARN</code></entry>
            <entry>Disabling exception sorter</entry>
          </row>
          <row>
            <entry>30002</entry>
            <entry><code>WARN</code></entry>
            <entry>Disabling exception sorter</entry>
          </row>
          <row>
            <entry>30003</entry>
            <entry><code>WARN</code></entry>
            <entry>Error checking exception fatality</entry>
          </row>
          <row>
            <entry>30004</entry>
            <entry><code>WARN</code></entry>
            <entry>Disabling validation connection checker</entry>
          </row>
          <row>
            <entry>30005</entry>
            <entry><code>WARN</code></entry>
            <entry>Disabling validation connection checker</entry>
          </row>
          <row>
            <entry>30006</entry>
            <entry><code>WARN</code></entry>
            <entry>Disabling stale connection checker</entry>
          </row>
          <row>
            <entry>30007</entry>
            <entry><code>WARN</code></entry>
            <entry>Disabling stale connection checker</entry>
          </row>
          <row>
            <entry>30008</entry>
            <entry><code>WARN</code></entry>
            <entry>HA setup detected</entry>
          </row>
          <row>
            <entry>30020</entry>
            <entry><code>WARN</code></entry>
            <entry>Detected queued threads during cleanup</entry>
          </row>
          <row>
            <entry>30021</entry>
            <entry><code>WARN</code></entry>
            <entry>Queued thread</entry>
          </row>
          <row>
            <entry>30022</entry>
            <entry><code>WARN</code></entry>
            <entry>Lock owned during cleanup</entry>
          </row>
          <row>
            <entry>30023</entry>
            <entry><code>WARN</code></entry>
            <entry>Lock is locked during cleanup without an owner</entry>
          </row>
          <row>
            <entry>30024</entry>
            <entry><code>WARN</code></entry>
            <entry>Error resetting transaction isolation</entry>
          </row>
          <row>
            <entry>30025</entry>
            <entry><code>WARN</code></entry>
            <entry>Error during connection listener activation</entry>
          </row>
          <row>
            <entry>30026</entry>
            <entry><code>WARN</code></entry>
            <entry>Error during connection listener passivation</entry>
          </row>
          <row>
            <entry>30027</entry>
            <entry><code>WARN</code></entry>
            <entry>Destroying connection that is not valid, due to the following exception</entry>
          </row>
          <row>
            <entry>30028</entry>
            <entry><code>WARN</code></entry>
            <entry>Error notifying of connection error for listener</entry>
          </row>
          <row>
            <entry>30040</entry>
            <entry><code>WARN</code></entry>
            <entry>Closing a statement you left open, please do your own housekeeping</entry>
          </row>
          <row>
            <entry>30041</entry>
            <entry><code>WARN</code></entry>
            <entry>Error during closing a statement</entry>
          </row>
          <row>
            <entry>30042</entry>
            <entry><code>WARN</code></entry>
            <entry>Closing a result set you left open, please do your own housekeeping</entry>
          </row>
          <row>
            <entry>30043</entry>
            <entry><code>WARN</code></entry>
            <entry>Error during closing a result set</entry>
          </row>
          <row>
            <entry>30050</entry>
            <entry><code>WARN</code></entry>
            <entry>Error creating connection</entry>
          </row>
          <row>
            <entry>30051</entry>
            <entry><code>ERROR</code></entry>
            <entry>Unable to load undefined URLSelectStrategy</entry>
          </row>
          <row>
            <entry>30052</entry>
            <entry><code>ERROR</code></entry>
            <entry>Unable to load URLSelectStrategy</entry>
          </row>
          <row>
            <entry>30053</entry>
            <entry><code>ERROR</code></entry>
            <entry>Unable to load URLSelectStrategy</entry>
          </row>
          <row>
            <entry>30054</entry>
            <entry><code>WARN</code></entry>
            <entry>Error creating XA connection</entry>
          </row>
          <row>
            <entry>30055</entry>
            <entry><code>ERROR</code></entry>
            <entry>Unable to load undefined URLXASelectStrategy</entry>
          </row>
          <row>
            <entry>30056</entry>
            <entry><code>ERROR</code></entry>
            <entry>Unable to load URLXASelectStrategy</entry>
          </row>
          <row>
            <entry>30057</entry>
            <entry><code>ERROR</code></entry>
            <entry>Unable to load URLXASelectStrategy</entry>
          </row>
          <row>
            <entry>30060</entry>
            <entry><code>WARN</code></entry>
            <entry>Error checking state</entry>
          </row>
          <row>
            <entry>31000</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Resource Adapter does not support running in a non-managed environment</entry>
          </row>
          <row>
            <entry>31001</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error during loading reauth plugin</entry>
          </row>
          <row>
            <entry>31002</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error during loading connection listener plugin</entry>
          </row>
          <row>
            <entry>31003</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong kind of ConnectionRequestInfo</entry>
          </row>
          <row>
            <entry>31004</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>No matching credentials in Subject</entry>
          </row>
          <row>
            <entry>31010</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong connection handle to associate</entry>
          </row>
          <row>
            <entry>31011</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Still active locks</entry>
          </row>
          <row>
            <entry>31012</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to obtain lock</entry>
          </row>
          <row>
            <entry>31013</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Interrupted attempting lock</entry>
          </row>
          <row>
            <entry>31014</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Connection has been destroyed</entry>
          </row>
          <row>
            <entry>31015</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Error during reauthentication</entry>
          </row>
          <row>
            <entry>31016</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong credentials passed to getConnection</entry>
          </row>
          <row>
            <entry>31017</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You cannot set autocommit during a managed transaction</entry>
          </row>
          <row>
            <entry>31018</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You cannot set read only during a managed transaction</entry>
          </row>
          <row>
            <entry>31019</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You cannot commit during a managed transaction</entry>
          </row>
          <row>
            <entry>31020</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You cannot commit with autocommit set</entry>
          </row>
          <row>
            <entry>31021</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You cannot rollback during a managed transaction</entry>
          </row>
          <row>
            <entry>31022</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>You cannot rollback with autocommit set</entry>
          </row>
          <row>
            <entry>31023</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Invalid connection</entry>
          </row>
          <row>
            <entry>31030</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Not wrapper for</entry>
          </row>
          <row>
            <entry>31040</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Connection is not associated with a managed connection</entry>
          </row>
          <row>
            <entry>31041</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Connection handle has been closed and is unusable</entry>
          </row>
          <row>
            <entry>31042</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Method is not implemented by JDBC driver</entry>
          </row>
          <row>
            <entry>31050</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The result set is closed</entry>
          </row>
          <row>
            <entry>31060</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>The statement is closed</entry>
          </row>
          <row>
            <entry>31070</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Transaction cannot proceed</entry>
          </row>
          <row>
            <entry>31080</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>DriverClass is undefined</entry>
          </row>
          <row>
            <entry>31081</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>ConnectionURL is undefined</entry>
          </row>
          <row>
            <entry>31082</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to create connection from datasource</entry>
          </row>
          <row>
            <entry>31083</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Wrong driver class for this connection URL</entry>
          </row>
          <row>
            <entry>31084</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to create connection</entry>
          </row>
          <row>
            <entry>31085</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Unable to create connection from URL</entry>
          </row>
          <row>
            <entry>31086</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>No DriverClass specified for URL</entry>
          </row>
          <row>
            <entry>31087</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Failed to register DriverClass</entry>
          </row>
          <row>
            <entry>31088</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>DataSourceClass is undefined</entry>
          </row>
          <row>
            <entry>31089</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Failed to load datasource</entry>
          </row>
          <row>
            <entry>31090</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>LocalTransaction only</entry>
          </row>
          <row>
            <entry>31091</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Trying to begin a nested LocalTransaction</entry>
          </row>
          <row>
            <entry>31100</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Could not load connection properties</entry>
          </row>
          <row>
            <entry>31101</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>XADataSourceClass is undefined</entry>
          </row>
          <row>
            <entry>31102</entry>
            <entry><code>EXCEPTION</code></entry>
            <entry>Failed to load XA datasource</entry>
          </row>
        </tbody>
      </tgroup>
    </table>
  </section>

</appendix>
